bpf/process/bpf_process_event.h

```diff
@@ -629,6 +629,9 @@ __event_get_cgroup_info(struct msg_execve_event *msg,
 	int zero = 0, subsys_idx = 0;
 	struct tetragon_conf *conf;
 
+	/* Ensure that we clear up cgroup info at the beginning*/
+	memset(&msg->kube, 0, sizeof(struct msg_k8s));
```

Comment:
As commented in the code, it needs to be in the beginning so if we return with an error all msg->kube fields are already cleared, otherwise we may return old unrelated cgroup info of another process..., msg is in the heap.

Reply:
Ah that makes sense. How about we move it at the top level function, in event_execve? This way it would be obvious that memset will be called before we return?

Could you also add some additional information to the comment? For example:
> /*  Clear cgroup info at the beginning, so if we return early we do not pass previous data */

Reply:
> Ah that makes sense. How about we move it at the top level function, in event_execve? This way it would be obvious that memset will be called before we return?

hmm

The event_execve() should only return early if it can't allocate entry from maps, otherwise it _should always_ submit a perf event, even if it is not complete while at same time marking the errors in the ->flags for missing related info. What I mean is if we can't read namespaces, then we mark ->flags and we must continue to read the other related data.

Here the return is just in the cgroup context msg->kube , if we fail to read cgroup, we return making sure that it is cleared, it does not mean return and stop collecting data.

Moving this in the beginning will take it out of from its context, and even if we memset(event) the full object: is too big and it will fail to compile... having the memset close to its usage is best choice.

Sure I will add that comment to improve it, ok!

Reply:
Not sure I understand.

You seem to be zeroing msg->kube because it has invalid data (from a previous run, I presume). Is that correct?
My argument is that we should do this at the top-level function, so that it will always hold even if we add other code that returns before the memset is executed.

That is, if we do add code like, then I think we would still have the same issue. 
```
     if (some_new_condition_here)
            goto err;
     ...
    __event_get_task_info(...)
     ...
     
     err:
       tail_call(ctx, &execve_calls, 0);
```

Am I missing something?

Reply:
But we should not add code like this, doesn't really make sense to me. In this case we should not do tail_call nor send perf event at all as data in heap is bogus anyway...; we should just return 0 instead of that goto err.

There are no many conditions that we should return , unless -ENOMEM , otherwise for all other errors continue gather all data: memset or initialize values , and on errors set ->flag error

Reply:
I add more code comments to note the appropriate behavior and avoid possible mistakes in the future. PTAL